impossible level did not have enough heap. Recovered memory from other code so works now

bbkiwi opened the pull request:
Will still blow up if turn on MAZE_DEBUG_PRINT as this tips over edge

bbkiwi committed:
impossible level did not have enough heap. Made easier.
Will still blow up if turn on MAZE_DEBUG_PRINT as this tips over edge

thaeli commented:
Tested, confirmed this works. It's only a small difficulty bump over "Killer", is this the max? Earlier it worked w/ an even smaller maze size.

bbkiwi committed:
To fix running out of memory (due to other modes etc using more)
have lowered some parameters used by CC code (which is not needed
  but still lurking around). Also shortened bpm sample time for
  color movement which uses less memory.

bbkiwi committed:
astyle

bbkiwi commented:
This 'fixes' the problem by lowering the memory requirement of color shake (bpm submodes) and changing FIXBPERO and OCTAVES to lowest value possible. (This is for CC which is not being used). Other CC stuff could be removed, but I will leave it to others.  Please test color shake now. I've pushed restart in IMPOSSIBLE maze and not got it to blow up. NOTE if you turn on maze_printf it will blowup!

AEFeinstein commented on firmware/user/modes/mode_color_movement.c:
```diff
@@ -76,7 +76,7 @@
 #define S_TO_MS_FACTOR 1000
 #define TITLE_VISIBLE_DURATION 4000000
-#define BPM_SAMPLE_TIME 12000 // in ms
+#define BPM_SAMPLE_TIME 4000 // 12000 // in ms
```
@bbkiwi is there any visual difference when cutting this number by two thirds?

bbkiwi replied:
Yes graphs for bpm submodes not as smooth and leds more variable. Your PR is better and forgot this one. (Sad to see CC vanish)

AEFeinstein replied:
CC can always come back, that's not tremendously hard to do. It was hard to justify keeping all that code around when it did nothing this year.

AEFeinstein commented:
Handled more aggressively by #111, which reclaimed 12.5k of RAM.
